lib/Doctrine/DBAL/Driver/OCI8/OCI8Connection.php

```
@@ -151,7 +151,11 @@ public function lastInsertId($name = null)
             return false;
         }
 
-        OraclePlatform::assertValidIdentifier($name);
+        $identifiers = explode('.', $name);
+
+        foreach ($identifiers as $identifier) {
```

TBH not even sure why we validate here at all.... Why not let the DB fail invalid identifiers.